### ISSUE TITLE
ProductFeesV0 returns http code 400

### ISSUE THREAD

**Issue by jbaelo** [jlevers/selling-partner-api#701]
## Problem description:

Any endponit under ProductFeesV0 is returning http cdoe 400  "Missing objects [FeesEstimateRequest] in body parameters."

Affected endopints are:
- getMyFeesEstimateForSku
- getMyFeesEstimateForAsin
- getMyFeesEstimates

This is caused by Crescat\SaloonSdkGenerator\Traits\HasArrayableAttributes ignoring $attributeMap on specific class and returning class property instead of map

![image](https://github.com/jlevers/selling-partner-api/assets/33716791/e3d32357-e3e7-4ceb-a71a-657b61ad73ce)

On line 51 instead of $this->attributeMap[$name] we have to use static::$attributeMap[$name] in order to access specific class maps
## Error:

```
Bad Request (400) Response
{
  "errors": [
    {
      "code": "InvalidInput",
      "message": "This is an invalid input",
      "details": "Missing objects [FeesEstimateRequest] in body parameters."
    }
  ]
}
```

## Code

```php
$connector = SellingPartnerApi::make(/* ... */)->seller();
$response = $connector->catalogItemsV20220401()->productFeesV0();
```

**Comment by MickDev13** [comment]
Exact same issue on my side.

I'm calling it like this:
```
$response = $connector->productFeesV0();
$apiInstance->getMyFeesEstimates($body)
```

This is the result in debug mode:

```
Saloon Request (GetMyFeesEstimates) ->^ array:6 [
  "connector" => "SellingPartnerApi\Seller\SellerConnector"
  "request" => "SellingPartnerApi\Seller\ProductFeesV0\Requests\GetMyFeesEstimates"
  "method" => "POST"
  "uri" => "https://sellingpartnerapi-eu.amazon.com/products/fees/v0/feesEstimate"
  "headers" => array:6 [
    "Content-Type" => "application/json"
    "Accept" => "application/json"
    "X-AMZ-Date" => "20240607T143936Z"
    "User-Agent" => "jlevers/selling-partner-api/v6.0.6/php"
    "Host" => "sellingpartnerapi-eu.amazon.com"
    "X-AMZ-Access-Token" => "xxxxxxxxxx"
  ]
  "body" => "[{"idType":"SellerSKU","idValue":"13881719","feesEstimateRequest":{"marketplaceId":"A13V1IB3VIYZZH","priceToEstimateFees":{"listingPrice":{"currencyCode":"EUR","amount":100},"shipping":{"currencyCode":"EUR","amount":0},"points":null},"identifier":"xxx-commfee-20240607163920","isAmazonFulfilled":true,"optionalFulfillmentProgram":null}}]"
]
Saloon Response (GetMyFeesEstimates) ->^ array:3 [
  "status" => 400
  "headers" => array:12 [
    "Server" => "Server"
    "Date" => "Fri, 07 Jun 2024 14:39:36 GMT"
    "Content-Type" => "application/json"
    "Content-Length" => "185"
    "Connection" => "keep-alive"
    "x-amz-rid" => "xxxxxx"
    "x-amzn-RateLimit-Limit" => "0.5"
    "x-amzn-RequestId" => "xxxxxxx"
    "x-amz-apigw-id" => "xxxxx"
    "X-Amzn-Trace-Id" => "Root=1-66631ba8-24b56376254441f3"
    "Vary" => "Content-Type,Accept-Encoding,User-Agent"
    "Strict-Transport-Security" => "max-age=47474747; includeSubDomains; preload"
  ]
  "body" => """
    {\n
      "errors": [\n
        {\n
          "code": "InvalidInput",\n
          "message": "This is an invalid input",\n
          "details": "Missing objects [PriceToEstimateFees] in body parameters."\n
        }\n
      ]\n
    }
    """
]

```

**Comment by jlevers** [comment]
Thanks for flagging this – the fix will be released in v7.

**Comment by jlevers** [comment]
Released in `v7.0.0`.
